JavaScript/TypeScript.sublime-syntax

```diff
@@ -739,7 +739,7 @@ contexts:
             - ts-type-expression-begin
 
     - match: satisfies{{identifier_break}}
-      scope: keyword.operator.type.js
+      scope: keyword.operator.type.satisfies.js
```

Comment:
I am not very confident with regards to all those type evaluation related keywords such as `typeof`, `instanceof`, ... accross all syntaxes.

Java syntax specification calls related expressions `type comparison expression`.

As such, those keywords could all be grouped under `keyword.operator.type`. In general, all of them - like `in` - have just the meaning to check, whether an object is of certain type.

Thus `keyword.operator.logical` or - following Java's naming - `keyword.operator.comparison` would be a good base scope. Maybe `keyword.operator.comparison.type`, which would probably be better aligned with suggestions for scoping `in` and `as`.

Same applies for the keywords below.

Reply:
`satisfies` is a weird TS-specific operator that's basically an inline type check that can also inform inference. I would scope it either identically to `as` or using the same approach as `as`. They both are only used in the type system and have no runtime effect.